src/app/core/components/media-table/media-table.component.ts

```diff
@@ -0,0 +1,132 @@
+import { Component, Input } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import {
+  MediaSortColumn,
+  MediaRow,
+  MediaSort,
+} from '../../models/table.interface';
+import { FormControl, ReactiveFormsModule } from '@angular/forms';
+import { SortButtonComponent } from '../sort-button/sort-button.component';
+import { RouterLink } from '@angular/router';
+import { NgIconComponent, provideIcons } from '@ng-icons/core';
+import { ionClose } from '@ng-icons/ionicons';
+
+@Component({
+  selector: 'app-media-table',
+  standalone: true,
+  imports: [
+    CommonModule,
+    ReactiveFormsModule,
+    SortButtonComponent,
+    RouterLink,
+    NgIconComponent,
+  ],
+  templateUrl: './media-table.component.html',
+  styleUrls: ['./media-table.component.scss'],
+
+  viewProviders: [provideIcons({ ionClose })],
+})
+export class MediaTableComponent {
```

Comment:
I see that you had fun working on this. I see that it's working.
If you won't need much more what an advanced table might deliver than it's okay, otherwise
you've opened pandoras box as you will suffer from maintaining a pretty complex ui object like a table is.

On the other side the motivation to remove ag grid is great and if this fits for a great while, then perfect 🙌